src/constants/gbhw.asm

```diff
@@ -32,14 +32,18 @@ SERIAL   EQU 3
 JOYPAD   EQU 4
 
 ; Joypad buttons mask
-J_RIGHT  EQU 1 << 0             ;  1
-J_LEFT   EQU 1 << 1             ;  2
-J_UP     EQU 1 << 2             ;  4
-J_DOWN   EQU 1 << 3             ;  8
-J_A      EQU 1 << 4             ; 10
-J_B      EQU 1 << 5             ; 20
-J_SELECT EQU 1 << 6             ; 40
-J_START  EQU 1 << 7             ; 80
+J_RIGHT  EQU 1 << 0     ;  1
+J_LEFT   EQU 1 << 1     ;  2
+J_UP     EQU 1 << 2     ;  4
+J_DOWN   EQU 1 << 3     ;  8
+J_A      EQU 1 << 4     ; 10
+J_B      EQU 1 << 5     ; 20
+J_SELECT EQU 1 << 6     ; 40
+J_START  EQU 1 << 7     ; 80
+J_DIRECTION_MASK            EQU J_RIGHT | J_LEFT | J_UP | J_DOWN ; $0F: filter A, B, Select and Start out
+J_DIRECTION_HORIZONTAL_MASK EQU J_RIGHT | J_LEFT                 ; $03: only keep right and left
+J_DIRECTION_VERTICAL_MASK   EQU J_UP | J_DOWN                    ; $0C: only keep up and down
+J_AB_MASK                   EQU J_A | J_B                        ; $30: only keep A and B
```

Comment:
To be honest, in the actual code, I find:

```suggestion
  and (J_UP | J_DOWN)
```

to be easier to read than:

```suggestion
  and J_DIRECTION_VERTICAL_MASK
```

But if you feel having more constants is better, that's also fine with me.

Reply:
Ok was an idea. Was not sure about that. But I would remove the brackets. They add no value.

Reply:
replaced again